### PR TITLE
Added stat tracking to the parser

### PR DESCRIPTION
**Pull request by lexicalbits**
Get a little insight into the numbers &  times inside the parser by creating a small framework for emitting statistics to the parent software.

### CODE REVIEW

File: src/parser.coffee [hunk 1]
```diff
@@ -95,6 +95,42 @@ class Parser
...
+      'condition': 0,
+      'previous': 0,
+      'continue': 0,
+      'redirect': 0
+    }
+    stats   = {
+      # How many total lines
+      lines: 0
+      # How many lines of actual code
+      , codelines: 0
```
Comment (octopi) [inline src/parser.coffee:115]: Honestly not super sure about the necessarily of counting the actual lines to this kind of granularity. `lines` makes sense, but `codelines`, `objectlines`, `comments` less so

Do you think total size in bytes of the script would be useful as well?

File: src/parser.coffee [hunk 2]
```diff
@@ -400,6 +451,7 @@ class Parser
 
         when "<"
           # < Label
+          cmdtype = 'label'
```
Comment (octopi) [inline src/parser.coffee:456]: Should this go on the open label `>`?

**Review by octopi — CHANGES_REQUESTED** [review]

File: src/parser.coffee (hunk 1 above)
Comment (lexicalbits) [reply src/parser.coffee:115]: The line counters seem pretty low overhead, but if you don't think they're useful, I'll axe 'em.  Total size might actually be an expensive operation - you'd have to JSON.stringify the final AST, which might be non-trivial for something quartz size.

File: src/parser.coffee (hunk 2 above)
Comment (lexicalbits) [reply src/parser.coffee:456]: It's either or since they're a matched set, so long as it's not in both.

File: src/parser.coffee (hunk 1 above)
Comment (octopi) [reply src/parser.coffee:115]: Yeah, stringifying at the end isn't great, but don't you already get the actual text of each line already? Would adding `line.length` counter each time be too expensive?

**Review by octopi — APPROVED** [review]

**Comment by octopi** [comment]
LGTM

**Pull request merged**